### PR TITLE
Expose Simulcast Configuration

### PR DESCRIPTION
**Pull request by dingyishen-amazon**
## ℹ️ Description
Expose simulcast configuration to customers by adding an API for mobile SDK to set simulcast in video client.

### Issue #, if available
https://sim.amazon.com/issues/Chime-49405

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
    - [x] README update
    - [x] CHANGELOG update
    - [x] guildes update
- [ ] This change requires a dependency update
    - [x] Amazon Chime SDK Media
    - [ ] Other (update corresonding legal documents)

## 🧪 How Has This Been Tested?
Turn on front and back cameras at the same time(back camera through contentshare), then start recording When simulcast is enabled, the recording action will fail, otherwise it succeed.

### Additional Manual Test
- [ ] Pause and resume remote video
- [ ] Switch local camera
- [ ] Rotate screen back and forth

## 📱 Screenshots, if available
*provide screenshots/video record if there's a UI change in demo app*

#### Integration validation (required before release)

- [x] Unit tests pass
- [x] Build SDK against simulator with release options
- [ ] Build SDK against device with release options
- [ ] Build SDK against simulator with debug options
- [ ] Build SDK against device with debug options
- [ ] Test Demo against simulator with SDK (debug build) in workspace
- [x] Test Demo against device with SDK (debug build) in workspace
- [ ] Test DemoObjC against simulator with SDK (debug build) in workspace
- [ ] Test DemoObjC against device with SDK (debug build) in workspace
- [ ] Test Demo against simulator with SDK.framework (release build)
- [ ] Test Demo against device with SDK.framework (release build)
- [ ] Test DemoObjC against simulator with SDK.framework (release build)
- [ ] Test DemoObjC against device with SDK.framework (release build)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AmazonChimeSDK/AmazonChimeSDK/audiovideo/AudioVideoControllerFacade.swift [hunk 1]
```diff
@@ -45,6 +45,17 @@ import Foundation
...
+    /// Start local video with configurations and begin transmitting frames from an internally held `DefaultCameraCaptureSource`.
+    /// `stopLocalVideo` will stop the internal capture source if being used.
+    ///
+    /// Calling this after passing in a custom `VideoSource` will replace it with the internal capture source.
+    ///
+    /// This function will only have effect if `start` has already been called
+    ///
+    /// Parameter config: configurations of emitted video stream, e.g. simulcast
+    /// - Throws: `PermissionError.videoPermissionError` if video permission of `AVCaptureDevice` is not granted
+    func startLocalVideo(config: LocalVideoConfiguration) throws
```
Comment (hensmi-amazon) [inline AmazonChimeSDK/AmazonChimeSDK/audiovideo/AudioVideoControllerFacade.swift:57]: Do we really need this? I think its fine to expect builders to write the few lines needed to inject a custom source. Long term i don't think we want a version that internally creates a source.

File: AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/LocalVideoConfiguration.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    /// The flag to disable/enable simulcast, default to true
+    public var simulcastEnabled: Bool
+    
+    convenience override public init() {
+        self.init(simulcastEnabled: true)
+    }
+    
+    public init(simulcastEnabled: Bool) {
+        self.simulcastEnabled = simulcastEnabled
```
Comment (hensmi-amazon) [inline AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/LocalVideoConfiguration.swift:22]: Nit: I feel like these inits are unnecessary, and clunky as we add more optional configurations. Lets just set defaults on the class itself.

**Review by hokyungh — APPROVED** [review]

File: AmazonChimeSDK/AmazonChimeSDK/internal/contentshare/DefaultContentShareVideoClientController.swift [hunk 3]
```diff
@@ -43,6 +43,12 @@ import Foundation
     }
 
     public func startVideoShare(source: VideoSource) {
+        logger.info(msg: "Starting video share with video source")
+        startVideoShare(source: source, config: LocalVideoConfiguration())
+    }
+
+    public func startVideoShare(source: VideoSource, config: LocalVideoConfiguration) {
+        // ignore LocalVideoConfiguration because contentshare does not have simulcast
```
Comment (georgezy-amzn) [inline AmazonChimeSDK/AmazonChimeSDK/internal/contentshare/DefaultContentShareVideoClientController.swift:51]: nit: I think it's better to move this comment to `startContentShare()` in `ContentShareController.swift` as documentation comment, maybe also add it to `LocalVideoConfiguration.simulcastEnabled`.

File: AmazonChimeSDK/AmazonChimeSDK/audiovideo/AudioVideoControllerFacade.swift (hunk 1 above)
Comment (dingyishen-amazon) [reply AmazonChimeSDK/AmazonChimeSDK/audiovideo/AudioVideoControllerFacade.swift:57]: We also need this for maxbitrate, right?

Comment (hensmi-amazon) [reply AmazonChimeSDK/AmazonChimeSDK/audiovideo/AudioVideoControllerFacade.swift:57]: Our current internal SDK API is designed as `XXXLocalVideo(source, config)` (i.e. it doesn't support calling without a source. Maybe its reasonable to have that create an internal source to make it easy for onboarding, but it requires the audio video controller to couple to the device controller logic. Anyways its not a big deal here i guess, just one extra function.

**Pull request merged**